index.bs

```diff
@@ -10056,35 +10174,60 @@ interface AudioWorkletProcessor {
 
 <h5 id="AudioWorketProcessor-constructors">
 Constructors</h5>
+
 <dl dfn-type="constructor" dfn-for="AudioWorkletProcessor">
 	: <dfn>AudioWorkletProcessor(options)</dfn>
 	::
-		When the constructor for {{AudioWorkletProcessor}} is invoked, the following steps are performed:
+		When the constructor for {{AudioWorkletProcessor}} is invoked,
+		the following steps are performed on the <a>rendering thread</a>.
+
+		If any of these steps throws an exception, abort the rest of
+		steps and <a>queue a task</a> to fire an
+		<a href="https://www.w3.org/TR/html50/webappapis.html#the-errorevent-interface">ErrorEvent</a>
+		named <code>processorerror</code> at the {{AudioWorkletNode}}
+		on the <a>control thread</a> with the relevant information about
+		the error.
+
 		<div algorithm="AudioWorkletProcessor()">
-			1. Compare the value of <a href="https://www.ecma-international.org/ecma-262/6.0/#sec-built-in-function-objects">NewTarget</a>
-				with the <a href="https://www.ecma-international.org/ecma-262/6.0/#sec-execution-contexts">active function object</a>;
-				if the two are equal, throw a {{TypeError}}.
-
-				Note: This check prevents the invocation of the constructor
-				directly from JavaScript. The interface object may only
-				be called internally by the UA.
-
-			1. Initialize the {{AudioWorkletProcessor}} using the
-				contents of the {{AudioWorkletNode/AudioWorkletNode()/options!!argument}} dictionary that was
-				passed to the constructor for {{AudioWorkletNode}}. The
-				contents of this dictionary will have been serialized and
-				deserialized according to the algorithm for <a href=
-				"#instantiation-of-AudioWorkletNode-and-AudioWorkletProcessor"> instantiating an AudioWorkletProcessor</a>.
-
-				Note: The base implementation of {{AudioWorkletProcessor}}
-				does not examine the values in the {{AudioWorkletProcessor/AudioWorkletProcessor()/options!!argument}}
-				dictionary. However, processor classes that extend
-				{{AudioWorkletProcessor}} will use this dictionary to
-				initialize themselves based on the options relevant to
-				their specific node type.
-
-			1. Set {{[[node reference]]}} to `null` and {{[[callable process]]}} to
-				`false`.
+			This constructor is invoked by processing a control
+			message queued by the {{AudioWorkletNode()|constructor}}
+			of the associated {{AudioWorkletNode}}. This control
+			message carries the [=processor construction data=] that
+			consists of:
+			<var>nodeName</var>,
+			<var>serializedProcessorPort</var>,
+			<var>serializedOptions</var>,
+			and <var>node</var>.
+
+			1. If there is no [=processor construction data=]
```

Comment:
Is that actually possible? (No processor construction data)

Reply:
This is the resolution from https://github.com/WebAudio/web-audio-api/issues/1963#issuecomment-510955630. This is a "magic" that prevents the user-supplied code from constructing an AudioWorkletProcessor.